src/BizHawk.Client.EmuHawk/MainForm.Events.cs

```diff
@@ -615,15 +615,25 @@ private void ViewSubMenu_DropDownOpened(object sender, EventArgs e)
 		private void WindowSizeSubMenu_DropDownOpened(object sender, EventArgs e)
 		{
 			var windowScale = Config.GetWindowScaleFor(Emulator.SystemId);
-			foreach (ToolStripMenuItem item in WindowSizeSubMenu.DropDownItems)
+			foreach (ToolStripMenuItem item in WindowSizeSubMenu.DropDownItems.OfType<ToolStripMenuItem>())
 			{
-				item.Checked = (int) item.Tag == windowScale;
+				if (item.Tag is int itemScale)
+				{
+					item.Checked = itemScale == windowScale && Config.ResizeWithFramebuffer;
+				}
 			}
+			DisableResizeWithFramebufferMenuItem.Checked = !Config.ResizeWithFramebuffer;
 		}
 
 		private void WindowSize_Click(object sender, EventArgs e)
 		{
 			Config.SetWindowScaleFor(Emulator.SystemId, (int) ((ToolStripMenuItem) sender).Tag);
+			FrameBufferResized(forceWindowResize: true);
+		}
+
+		private void DisableResizeWithFramebufferMenuItem_Click(object sender, EventArgs e)

```

Comment:
I think if you inserted this method above the other one the diff would be cleaner? Not that important